### PR TITLE
Add Code coverage reporting to crossplane project.

### PR DESCRIPTION
**Pull request by ichekrygin**
Add code coverage reporting to the project:
- CICD: leveraging Cobertura reports 
image
image
- GitHub PR: leveraging `codecov.io` integration

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -81,6 +83,9 @@ pipeline {
                         }
                     }
                 }
+                script {
+                    sh 'curl -s https://codecov.collibra.com/bash | bash -s -- -c -f _output/tests/**/coverage.txt -F unittests'
```
Comment (negz) [inline Jenkinsfile]: It this URL any different from https://codecov.io/bash? I haven't seen it before.

File: Jenkinsfile [hunk 2]
```diff
@@ -61,6 +62,7 @@ pipeline {
                 always {
                     archiveArtifacts "_output/tests/**/*"
                     junit "_output/tests/**/unit-tests.xml"
+                    cobertura autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: '_output/tests/**/coverage.xml', enableNewApi: false, failUnhealthy: false, failUnstable: false, maxNumberOfBuilds: 0, onlyStable: false, sourceEncoding: 'ASCII', zoomCoverageChart: false, classCoverageTargets: '50, 0, 0', conditionalCoverageTargets: '70, 0, 0', lineCoverageTargets: '40, 0, 0', methodCoverageTargets: '30, 0, 0', packageCoverageTargets: '80, 0, 0'
```
Comment (negz) [inline Jenkinsfile]: This might be a little more readable broken over many lines, especially given the coverage targets are way at the end.

**Comment by negz** [comment]
@ichekrygin I notice this PR also seems to be 'updating' the build submodule with changes that are already in place, as far as I can tell.

File: Jenkinsfile (hunk 1 above)
Comment (ichekrygin) [reply Jenkinsfile]: I got this from the codecov docs site: https://docs.codecov.io/docs/supported-ci-providers.

**Pull request merged**